File: frontend/types/operations.ts
export interface OperationsEvent {
    event_id: string;
    event_type: string;
    lat: number;
    lon: number;
    confidence: number;
    priority: string;
    timestamp: string;
    aoi_name?: string;
}

export interface SiteStatus {
    id: string;
    name: string;
    type: string;
    priority: string;
    country: string;
    today_count?: number | null;
    baseline?: number | null;
    anomaly_factor?: number | null;
    status: "normal" | "elevated" | "anomalous" | string;
    today_flights?: number | null;
    flight_baseline?: number | null;
    flight_anomaly?: "normal" | "elevated" | "anomalous" | string;
}

export interface SiteProperties {
    id: string;
    name: string;
    country: string;
    type: string;
    lat: number;
    lon: number;
    bbox: [number, number, number, number];
    priority: string;
    tags?: string[];
}

export interface SiteFeature {
    type: "Feature";
    geometry: {
        type: "Point";
        coordinates: [number, number];
    };
    properties: SiteProperties;
}

export interface SiteGeoJson {
    type: "FeatureCollection";
    features: SiteFeature[];
}

export interface IntelArticle {
    title?: string;
    url?: string;
    source?: string;
    source_tier?: number;
    published_at?: string;
    site_id?: string | null;
}

export interface SiteIntelResponse {
    site_id: string;
    articles: IntelArticle[];
    hours: number;
}

export interface FlightState {
    site_id?: string | null;
    icao24: string;
    callsign?: string | null;
    origin_country?: string | null;
    lat?: number | null;
    lon?: number | null;
    altitude_m?: number | null;
    velocity_ms?: number | null;
    heading?: number | null;
    on_ground: boolean;
    timestamp: string;
}

export interface FlightActivity {
    site_id: string;
    recent_count: number;
    unique_aircraft: number;
    on_ground_count: number;
    airborne_count: number;
    latest_states: FlightState[];
}

File: backend/app/schemas/operations.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class CountResponse(BaseModel):
    count: int


class OperationsEvent(BaseModel):
    event_id: str
    event_type: str
    lat: float | None = None
    lon: float | None = None
    confidence: float | None = None
    priority: str
    timestamp: datetime
    aoi_name: str | None = None


class AoiBaselineResponse(BaseModel):
    aoi_id: str
    event_type_baselines: dict[str, float]
    data_points: int
    lookback_days: int


class SiteStatusResponse(BaseModel):
    id: str
    name: str
    type: str
    priority: str
    country: str
    today_count: int | None = None
    baseline: float | None = None
    anomaly_factor: float | None = None
    status: str
    today_flights: int = 0
    flight_baseline: float = 0.0
    flight_anomaly: str = "normal"


class FlightStateResponse(BaseModel):
    site_id: str | None = None
    icao24: str
    callsign: str | None = None
    origin_country: str | None = None
    lat: float | None = None
    lon: float | None = None
    altitude_m: float | None = None
    velocity_ms: float | None = None
    heading: float | None = None
    on_ground: bool = False
    timestamp: datetime


class FlightActivityResponse(BaseModel):
    site_id: str
    recent_count: int
    unique_aircraft: int
    on_ground_count: int
    airborne_count: int
    latest_states: list[FlightStateResponse]


class IntelArticleResponse(BaseModel):
    title: str
    url: str
    source: str | None = None
    source_tier: int
    published_at: datetime | None = None
    site_id: str | None = None
